conjure_oxide/tests/generated_tests.rs

```diff
@@ -150,210 +222,266 @@ fn integration_test_inner(
+        // Perform the assertion immediately
+        assert_eq!(
+            expected, generated,
+            "Generated rule trace does not match the expected trace!"
+        );
 
-    let generated_json_rule_trace = read_rule_trace(path, essence_base, "generated", accept)?;
-    let expected_json_rule_trace = read_rule_trace(path, essence_base, "expected", accept)?;
+        (Some(generated), Some(expected))
+    } else {
+        (None, None) // Avoid uninitialized variables when 4a is disabled
+    };
 
-    assert_eq!(expected_json_rule_trace, generated_json_rule_trace);
+    if accept {
+        // Overwrite expected parse and rewrite models if enabled
+        if config.enable_native_parser {
+            model_native.clone().expect("model_native should exist");
+            copy_generated_to_expected(path, essence_base, "parse", "serialised.json")?;
+        }
+        if config.parse_model_default {
+            copy_generated_to_expected(path, essence_base, "parse", "serialised.json")?;
+        }
+        if config.apply_rewrite_rules {
+            copy_generated_to_expected(path, essence_base, "rewrite", "serialised.json")?;
+        }
 
-    let generated_rule_trace_human =
-        read_human_rule_trace(path, essence_base, "generated", accept)?;
-    let expected_rule_trace_human = read_human_rule_trace(path, essence_base, "expected", accept)?;
+        if config.solve_with_minion {
+            copy_generated_to_expected(path, essence_base, "minion", "solutions.json")?;
+        }
 
-    assert_eq!(expected_rule_trace_human, generated_rule_trace_human);
+        if config.validate_rule_traces {
+            copy_human_trace_generated_to_expected(path, essence_base)?;
+            save_stats_json(context.clone(), path, essence_base)?;
+        }
+    }
 
-    // Stage 4: Run the model through the Minion solver and check that the solutions are as expected
-    if config.run_solver {
-        // TODO: when we do the big refactor, lump all these pass-through variables into a state
-        // struct
-        check_solutions_stage(
-            &context,
-            model,
-            path,
-            essence_base,
-            extension,
-            verbose,
-            accept,
-        )?;
+    // Check Stage 1b (native parser)
+    if config.enable_native_parser {
+        let expected_model = read_model_json(&context, path, essence_base, "expected", "parse")?;
+        let model_native = model_native.expect("model_native should exist here");
+        assert_eq!(model_native, expected_model);
     }
 
+    // TODO (yb33): Investigate: for some reason, model.expect(), and model from "expected" file aren't identical.
+    // Check Stage 1a (parsed model)
+    if config.parse_model_default {
+        let expected_model = read_model_json(&context, path, essence_base, "expected", "parse")?;
+        let model_from_file = read_model_json(&context, path, essence_base, "generated", "parse")?;
```

Comment:
@niklasdewally: around here instead of model_from_file, using model.expect("") should recreate the issue. 

@YehorBoiar: is that right?

Reply:
Was there a particular parser that this was broken with?

Reply:
You mean test case? @YehorBoiar will have to remember / dig it up, I don't remember which one it was, sorry.

Reply:
Is this bug specific to the treesitter parser or the json parser?

Reply:
(if thats unknown, I'll try both)

Reply:
I don't know.

Reply:
Should model.clone() deep clone or shallow clone a model?

I think the symbol tables are staying linked on clone, making machine names appear in the parsed model output...